### ISSUE TITLE
Resolve reports of all activated accounts not loading by default

### ISSUE THREAD

**Issue by 314159265359879** [blockstack/blockstack-app#3038]
https://github.com/hirosystems/stacks-wallet-web/issues/3020 revealed this issue when GAIA is down and the fallback isn't working that has caused some issues where a user would be left with only the first account showing, even after the fix https://github.com/hirosystems/stacks-wallet-web/pull/3021 (in rare cases). 

1. Gaia stores the number of accounts created. 
2. The fallback is a local store of the value. 
3. The fix will check accounts for activity until it finds 20 without activity.

It looks like when gaia is down, the wallet will default to one account and in the process update the locally stored value. 

Currently the number of accounts on a key is only checked when the wallet is first restored. Which means the issue of seeing a single wallet will persist if it fails at the start. 

Some thoughts to solve the problem for users:
1. Do an automatic check repeatedly (i.e. daily) 
2. Have a button to restore addresses
3. Have an option for users to restore a specific number of accounts (such as with debug.setHighestAccountIndex() in the debug tools)

Although this happens rarely a user experiencing "lost" accounts is often worried about if/how to get them back. Currently the interface shows "Create an Account" is this the best wording? is it even being created? ;-) What about if we call it "Create/restore an Account", or ditch "create" completely and call it "Reveal an Account", "Reveal an extra account". Ledger uses "Add account", nice and short? 

On a wallet with 1 account only "Create an Account" is shown in the overflow menu and not "switch account" what about showing "Restore Accounts" instead here perhaps with a little tooltip. "If you created accounts before but do not see them, you can use this button to restore those accounts."
![image](https://user-images.githubusercontent.com/33360391/215757279-f640ebd9-9ddc-45b9-b8b0-a1850e179de4.png)

**Comment by 314159265359879** [comment]
If restoring the wallet from the seed does not reveal all your accounts for you. 

- Then you can use "Create an Account" in the top right wallet menu and it will reveal the accounts again.

- Additionally these steps can speed up the process of revealing the accounts (wallet version 3.32.2 and up), especially helpful if you have many:

1. Right click the extension icon 
2. Open Hiro Wallet in a new tab
3. Open console: Ctrl+Shift+I
4. copy this: debug.setHigestAccountIndex(XX)
5. replace "XX" with the number of accounts you want to reveal (minus 1)
6. press `<Enter>`


What that could look like
![image](https://user-images.githubusercontent.com/33360391/216555580-a93a7f65-385b-4533-85a6-5272db30f7a6.png)

![image](https://user-images.githubusercontent.com/33360391/216555984-bddafc11-20d3-49b0-a51a-000154c66c15.png)

**Comment by markmhendrickson** [comment]
@314159265359879 will provide further recreation info here soon for @kyranjamie to take a look

**Comment by 314159265359879** [comment]
TLDR:
* It may still be best to wait what happens after Chrome store is up to date. I believe some issues look more persistent  because the Chrome version isn't up to date. 
* Perhaps checking for activity on unrevealed addresses after unlock in addition to the current check when restoring from seed would prevent issues from persisting after the wrong number of accounts is shown? 
* Should debug.setHighestAccountIndex or the automatic check for activity change `localstore`?
* Check for FT, NFT and BNS balance as well as transaction activity from the wallet. That way you catch accounts like this (18):   https://explorer.stacks.co/address/SP2PCFCT6R6E8PB9JW92YR6G7S7RXTR9B0YQXPS70?chain=mainnet

............... Firefox

When I have a fresh installation of Firefox and the wallet (3.32.3) and I restore from the seed I will see (the switch account menu not show) or show a limited number of accounts (three accounts here, after using "Create an account" for the second time). 
Then there is some loading (see at 44 sec.) and all the accounts with activity will show. (17) Adjusting with the debug option has no lasting influence, should it? (I can adjust the index to 10 or 20, it will normally show 17 after restoring from seed). 

https://user-images.githubusercontent.com/33360391/217006783-09d3878f-be21-4121-aaa9-bf8844326839.mp4

After many repetitions of log out and restoring from seed there was one instance when only the first three accounts would show (are these retrieved from the local stored number of accounts? I have not been able to reproduce this on video. 

Once I have clicked "create an account" enough times ca 10-16 times it will simply show all the accounts whenever you restore from seed. There is no blinking/loading accounts (I assume because no activity detected on unrevealed accounts) after that and every sign out, and restore from seed will show the same:

https://user-images.githubusercontent.com/33360391/217006721-04038115-afab-41db-8ecd-54498a318d6d.mp4

...................... Chrome 

The chrome store has not yet updated to 3.32.3, I believe that is the reason some users may still be experiencing the issue (missing accounts). If gaia was not available at some time in the past when the wallet updated or was restored from seed it would not show all the accounts. When I use 3.31.1 and block the gaia connection, I see only one account loading on a fresh browser/wallet install. There is no button to switch. 
![image](https://user-images.githubusercontent.com/33360391/217009756-afd7c6f9-7b74-4dcb-8011-087d2221d381.png)

Then if the gaia connection is restored, the wallet still loads with the limited number of accounts after lock/unlock. Additionally I experienced issue #3032 during these tests (diagnostics allowed). 

...................... Account holding just one NFT

Account 18 ( SP2PCFCT6R6E8PB9JW92YR6G7S7RXTR9B0YQXPS70 ) is not loading (see first video) due to activity, it has no transactions send from that address, but it does hold an NFT. I had a user reach out mentioning they were missing an account with some NFT's perhaps this was a similar situation. Could we add a check for available tokens (FT, NFT, BNS as well as activity)?
